### DISCUSSION TITLE
Data Export, Column Grouping and Column Hiding

### DISCUSSION THREAD

**Discussion by dbnar2** [icflorescu/mantine-datatable#149] [Q&A]
First of all apologies if this is too much for one post but I did not feel it was necessary to create different Discussion topics around this.

Love this component and all the new features that get added. Wanted to ask 2 main things:

1) Does the library support the ability to export Data for example if we wanted to export to CSV or Excel?

2) Are we able to create column groups?
Example: 
![Screenshot_20221214_144604](https://user-images.githubusercontent.com/36969747/207748704-8dd54270-ed99-4508-80fd-df8e5fe36961.jpg)

Any ideas on how we can implement this would be great! Still a junior developer but wouldnt mind contributing if pointed in the right direction!

**Comment by dbnar2** [comment]
Updated Post. Just saw that row properties includes hidden so that's already taken care of.
Just missing the data export and row groupings

**Comment by icflorescu** [comment]
Exporting data has been requested and discussed before (see https://github.com/icflorescu/mantine-datatable/discussions/122).
All things considered, it falls outside the scope of this project.

**Reply by dbnar2** [reply]
Understood. Just to be clear I was not suggesting a full on export button within the table. Thay would create too much bloat. I was mainly suggesting of a hook to get the data (json array) of what is currently being displayed and then I could implement my own export either using exceljs or csvwriter etc.

Also what about the idea of the column groupings? Do you feel that falls out of scope of the project?
